packages/kbn-config-schema/src/types/maybe_type.test.ts

```diff
@@ -60,3 +60,30 @@ test('includes namespace in failure', () => {
   const type = schema.maybe(schema.string());
   expect(() => type.validate(null, {}, 'foo-namespace')).toThrowErrorMatchingSnapshot();
 });
+
+describe('maybe + object', () => {
+  test('returns undefined if undefined object', () => {
+    const type = schema.maybe(schema.object({}));
+    expect(type.validate(undefined)).toEqual(undefined);
+  });
```

Comment:
The way I encountered the bug was with given definition:

```typescript
const type = schema.maybe(
            schema.object({
              type: schema.string(),
              id: schema.string(),
            })
          )
```

which is not validating when calling `type.validate(undefined)`. I know the `applyDefaults` is tested on the `object` type side, but I don't see the `maybe` + `applyDefaults` case covered?

Reply:
I have added a test for your example, but it is not fundamentally different than the test here. `schema.maybe` should allow `undefined` regardless of what the underlying type is.

Reply:
Yes, but the way `schema.maybe` works internally is by altering the nested's `JOI` type declaration, that's why I thought that kind of test probably makes sense.